.github/workflows/build_and_test.yml

```diff
@@ -13,42 +13,39 @@ concurrency:
 jobs:
   build-and-test:
     runs-on:
-      labels: ubuntu-22.04-32core
+      labels: ubuntu-24.04-64core
     steps:
-      - name: Install dependencies
-        run: |
-          sudo apt-get install -y libomp-dev
       - name: Check out repository code
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # pin@v3
 
+      - name: Install dependencies
+        run: |
+          ./.github/install_clang_version.sh 19
+
       - name: Cache bazel build artifacts
         uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # pin@v4.2.0
         with:
           path: |
             ~/.cache/bazel
-          key: ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-${{ hashFiles('bazel/import_llvm.bzl') }}
+          key: ${{ runner.os }}-clang19-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-${{ hashFiles('bazel/import_llvm.bzl') }}
           restore-keys: |
-            ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-${{ hashFiles('bazel/import_llvm.bzl') }}
-            ${{ runner.os }}-bazel-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-
+            ${{ runner.os }}-clang19-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-${{ hashFiles('bazel/import_llvm.bzl') }}
+            ${{ runner.os }}-clang19-${{ hashFiles('.bazelversion', '.bazelrc', 'WORKSPACE.bzlmod', 'MODULE.bazel') }}-
+
+      - name: Confirm bazel is using the right C compiler
+        run: |
+          # Pure debugging bazel stuff: ensure the compiler installed is the
+          # one used by bazel. Note that the path to the C compiler is part of
+          # the cache, so if the compiler changes but the cache does not, it
+          # will try to use the old compiler path.
+          which -a clang | xargs ls -al
+          ls -al /etc/alternatives/clang
+          bazel cquery --output=starlark --starlark:expr="str(providers(target))" @bazel_tools//tools/cpp:current_cc_toolchain | sed 's/,/\n/g' | grep -C 5 clang
 
       - name: "Run `bazel build`"
         run: |
-          bazel build -c opt //... --explain=bazel-explain.log --verbose_explanations
+          bazel build -c opt //...
```

Comment:
why not cache these builds?

Reply:
They are cached. It's weird because the earlier cache action sets up a post-action cache save when there's a cache miss. The action below that was removed is a forced cache save that errs when the cache key is present. I was earlier experimenting with trying to override the github cache because I wasn't sure it was working properly and I was kind of fumbling.